x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx

```diff
@@ -29,16 +38,49 @@ export function ApmServiceContextProvider({
 }) {
   const { urlParams } = useUrlParams();
   const { agentName } = useServiceAgentNameFetcher();
+
+  const { start, end, environment } = urlParams;
+  const { serviceName } = useParams<{ serviceName?: string }>();
+
   const transactionTypes = useServiceTransactionTypesFetcher();
+
   const transactionType = getTransactionType({
     urlParams,
     transactionTypes,
     agentName,
   });
 
+  const { data: { alerts } = { alerts: [] as APMServiceAlert[] } } = useFetcher(
+    (callApmApi) => {
+      if (!start || !end || !serviceName || !transactionType) {
+        return;
+      }
+
+      return callApmApi({
+        endpoint: 'GET /api/apm/services/{serviceName}/alerts',
+        params: {
+          path: {
+            serviceName,
+          },
+          query: {
+            start,
+            end,
+            transactionType,
+            environment,
+          },
+        },
+      });
+    },
+    [start, end, serviceName, transactionType, environment]
+  );
```

Comment:
Can you extract this similar to `useServiceTransactionTypesFetcher` and `useServiceAgentNameFetcher`?

```suggestion
const alerts = useApmServiceAlertsFetcher()
```

Reply:
Not now, but we should also fetch the existing rules for a service, so we can show an "edit alert" button if one already exists in addition to add alert.

Reply:
I've moved the alerts fetching to `useServiceAlertsFetcher`.